### DISCUSSION TITLE
Docker image

### DISCUSSION THREAD

**Discussion by stefanvanherwijnen** [vendurehq/vendure#2231] [General]
Hi,

I would like to use Vendure as an ecommerce backend for a project (Node server + SSR rendered web app) I am working on. The project runs in a Docker image which I configure with Docker Compose.
I would like to run the Vendure image inside the same Docker Compose file, but there doesn't seem to be an official Vendure image.
It's been a while since I used Vendure, but iirc the build depends on the configuration which would explain why one would need to build the Docker image using the Dockerfile.

Would it be possible to have a official Vendure Docker image which can entirely be configured with environment variables and volume mappings? Perhaps some kind of base image which would be loaded in the Dockerfile:
```
# Something like this
FROM vendure

WORKDIR /usr/src/app
COPY vendure.config.ts .
```

I'm hoping someone has some thoughts on this.

**Comment by michaelbromley** [comment]
Hi Stefan,

Thanks for the suggestion. The thing with Vendure is that its strength comes from configurability. With an official docker image, how would we handle the situation where a user needs to, e.g. define custom fields?

Some things can be easily done via env vars like what port to use etc, but complex objects as in custom fields would need to be stored in a file in any case. Same goes for configuration of plugins.

I think in the end you'd need your own vendure config file which would then be used inside the container. As soon as you start building a custom plugin (which is the case in close to 100% of real projects), you then need to copy all of the plugin files into the container.

At that point the value of an "official" image comes into question, and the better option seems to be to provide you with a basic Dockerfile (as we already do) which you can then point to from your Docker Compose file.

**Reply by stefanvanherwijnen** [reply]
Thanks Michael,

That was indeed what I was thinking. Perhaps the best way is to create my own Vendure project and publish it myself (e.g. a my-project-vendure image). The main problem with building from source (every time) is that it can take a while and that minor updates tend to break the build process (usually just pulling the latest dependency versions from npm manages to break something :disappointed: )

I'll see if I can come up with a clever solution and I'll get back here if I do.
